sort/mergesort.go

```diff
@@ -60,3 +60,29 @@ func MergeIter[T constraints.Ordered](items []T) []T {
 	}
 	return items
 }
+
+// MergeParallel Perform merge sort on a slice using goroutines
+func MergeParallel[T constraints.Ordered](items []T) []T {
+
+	if len(items) < 2 {
+		return items
+
+	}
+
+	if len(items) < 2048 {
```

Comment:
Your implementation is never tested.

Reply:
Hi ! 

Indeed this conditions has been added to prevent having too many goroutines.

Would you prefer me to remove it even if it might cause performance issues or should I add a specific test case using a bigger array like this one : 
```go
	t.Run("MergeParallel on big array", func(t *testing.T) {
		rand.Seed(time.Now().UnixNano())
		size := 100000
		randomBigArray := make([]int, size)
		for i := range randomBigArray {
			randomBigArray[i] = rand.Intn(size)
		}

		sortedArray := sort.MergeParallel[int](randomBigArray)
		for i := 0; i < len(sortedArray)-1; i++ {
			if sortedArray[i] > sortedArray[i+1] {
				t.Errorf("MergeParallel failed")
			}
		}
	})
```

Reply:
Add a specific test rather than sticking to the framework in the test package for sorting tests.

Reply:
Fixed in [8ad3642](https://github.com/TheAlgorithms/Go/pull/558/commits/8ad364295e212aa906d1e061e85a688d7a89a13a)